src/components/SocialMediaLinks/SocialMediaLinks.tsx

```diff
@@ -0,0 +1,195 @@
+  size?: SizeType;
+}
+
+/* Accepts an array containing one or more data objects, each representing a social media platform and optional overrides for the label and/or URL.
+ *
+ *  @returns an array of data objects for each requested platform type that includes type, iconName, labelText and url where the latter two may include values supplied to override the defaults.
+ */
+function getLinksData(platforms: SocialMediaLinkDataProps[]) {
+  let allData = [];
+  platforms.forEach((myPlatform) => {
+    // Test if the type is allowed
+    if (!socialMediaLinkTypeArray.includes(myPlatform.type)) {
+      console.error(
+        "NYPL Reservoir SocialMediaLinks: " +
+          myPlatform.type +
+          " is either not a supported platform or is misspelled.\n" +
+          "Supported platforms include: " +
+          socialMediaLinkTypeArray.join(", ")
+      );
+      return;
+    }
+
+    // Get the dataset for this platform.
+    let thisPlatformArray = socialMediaDataMap.filter(
+      (socialMediaSlice) => socialMediaSlice.type === myPlatform.type
+    );
+
+    // The filter function returns an array of objects, so let's just get the objects out of the array.
+    let thisPlatformData = thisPlatformArray[0];
+
+    // If a url value exists in myPlatform use it, else use the default.
+    let newUrl = myPlatform.url ?? thisPlatformData.url;
+
+    // If a labelText value exists in myPlatform use it, else use the default.
+    let newLabelText = myPlatform.labelText ?? thisPlatformData.labelText;
+
+    let thisObj = {
+      type: thisPlatformData.type,
+      iconName: thisPlatformData.iconName,
+      labelText: newLabelText,
+      url: newUrl,
+    };
+    allData.push(thisObj);
+  }); // end foreach
+
+  return allData;
+}
+
+/**
+ * The SocialMediaLinks component renders a list of links for accessing social media sites.
+ */
+export const SocialMediaLinks = chakra(
+  forwardRef<HTMLDivElement, SocialMediaLinksProps>((props) => {
+    const {
+      borders = "none",
+      color = "textDefault",
+      className,
+      id,
+      linksData,
+      layout = "row",
+      showLabels = false,
+      size = "medium",
+      ...rest
+    } = props;
+
+    // Turns out you can pass whatever props you want to this thing in order to do logic in the theme.
+    const styles = useMultiStyleConfig("SocialMediaLinks", {
+      variant: borders,
+      size: size,
+      color: color,
+      layout: layout,
+    });
+
+    if (showLabels && borders === "circular") {
+      console.error(
+        "NYPL Reservoir SocialMediaLinks: `showLabels` is set to true, but labels are disallowed when also using borders='circular'."
+      );
+      return (
+        <p>
+          <Icon name={"alertWarningFilled"} size={"xlarge"} color={"red"} />
+          Error: See Console Log
+        </p>
+      );
+    }
+
+    // If linksData has values, use them, else use the entire list of platforms.
+    const socialMediaDataArray = linksData
```

Comment:
According to the spec doc, `linksData` is a required prop. Instead of using the entire list of platforms, you can log this message "NYPL Reservoir SocialMediaLinks: The `linksData` prop is required." and  `return null` so that nothing is rendered.

Reply:
In subsequent conversation w/ Marty it was decided to make linksData optional and return all platforms if none is provided.

Reply:
@bigfishdesign13 could we make an addendum or edit the spec doc?

Reply:
It was left that I would update the docs when I wrote them. No need to update the spec doc from my point of view.

Reply:
I will update the spec doc to reflect the change to the `linksData` prop and how the default will be handled.